opflexagent/utils/bridge_managers/vpp_manager.py

```diff
@@ -0,0 +1,143 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from neutron.plugins.ml2.drivers.openvswitch.agent.common import constants
+from opflexagent.utils.bridge_managers import bridge_manager_base
+from opflexagent.utils.bridge_managers import trunk_skeleton
+from oslo_config import cfg
+from oslo_log import log as logging
+from vpplib.VPPApi import VPPApi
+
+
+LOG = logging.getLogger(__name__)
+
+
+class VppManager(bridge_manager_base.BridgeManagerBase,
+                 trunk_skeleton.OpflexTrunkMixin):
+    """ Bridge Manager for VPP."""
+
+    def __init__(self):
+        super(VppManager, self).__init__()
+
+    def initialize(self, host, opflex_conf, agent_state):
+        self.int_br_device_count = 0
+        try:
+            vpp_config = cfg.CONF.VPP
+        except cfg.NoSuchOptError:
+            raise
+        agent_state['vhostuser_socket_dir'] = vpp_config.vhostuser_socket_dir
+        return self, agent_state
+
+    def get_local_ip(self):
+        return None
+
+    def check_bridge_status(self):
+        vapi = VPPApi(LOG, 'gbp-agent')
+        version = vapi.get_version()
+        if (version['retval'] == 0) and version['version']:
+            return constants.OVS_NORMAL
+        return constants.OVS_DEAD
+
+    def setup_integration_bridge(self):
+        """Override parent setup integration bridge.
+        VPP renderer handles the bridge creation.
+        Nothing to do here
+        """
+        pass
+
+    def scan_ports(self, registered_ports, updated_ports=None):
```

Comment:
This method seems identical to the one in the ovs bridge manager. Maybe we should move such methods to a common parent class

Reply:
In general, how a datapath scans for ports should be datapath specific. It still does not feel right to move this to the main bridge manager class. Let me know if you disagree

Reply:
For the sake of reusability, we can have default behaviors that each datapath plugin can override if needed.

Reply:
Sure, but since this is the only function as of now, that has common code, as we discussed, can be continue for now. Agree that if more such functions come up, we should create a mixin class.